File: python/3d_ex/main.py
def err(exc):
    raise Exception(exc)

def init():
    return int(input())

def counting_dup(n):
    a = b = init()
    count = 0

    for _ in range(n - 1):
        a = init()
        if a == b:
            count += 1
        b = a
    return count

n = init()
if n <= 0:
    err("Invalid input")

print(counting_dup(n))

File: kotlin/3d_ex/main.kt
fun err(exc: String): Nothing {
    throw Exception(exc)
}

fun init(): Int = readLine()?.toIntOrNull() ?: err("Invalid input")

fun countingDup(n: Int): Int {
    var count = 0
    var a = init()
		var b = a
    for (i in 1 until n) {
        a = init()
        if (a == b) {
            count++
        }
        b = a
    }
    return count
}

fun main() {
    val n = init()
    if (n <= 0) {
        err("Invalid input")
    }

    println(countingDup(n))
}
